### PR TITLE
Connect with Bookstore api

### PR DESCRIPTION
**Pull request by chiztechnology**
In the previous project, we developed the add book and remove book features directly in our application using Redux and React state. Now we just refactored our code to use the Bookstore API to add and remove books.

### Project requirements

1. Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API. ✅ 
2. Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls. ✅ 
3. Fetch book list from Bookstore API on Books page load. ✅ 

- Add an Action Creator, that returns a function. ✅ 
- Fetch list of books. ✅ 
- Dispatch another action (add an action type) that will pass API response to reducer and updates the state. ✅ 

4. Refactor your add book and remove book features to persist your changes in the server. ✅ 

- Change your existing Action Creators, so they return functions. ✅ 
- Make changes by using API. ✅ 
- Dispatch an action (action type that existed before refactor) that will updates the state accordingly. ✅ 

5. Add book and remove book should work in the same way after the refactor. ✅ 

6. No styling is required. ✅

### CODE REVIEW

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @chiztechnology, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

<div align="center">
  OopsNextGIF

</div>

You have submitted a project having a modified linters configuration. Kindly ensure that your `.eslintrc.json` matches [that of the linters configuration for react/redux projects](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by chiztechnology** [comment]
Hello dear @emmiiorji . Thank you for your great review.

 But I’m getting this error using that eslintrc config file . I don’t get what’s wrong with the file. 
 
Kindly check and let me know 

6B026429-1ADB-4B96-A419-2404C96EB929

**Comment by emmiiorji** [comment]
Hi @chiztechnology

Sorry about. I also see that you're using the incorrect `linters.yml` file. You should be using [this](https://github.com/microverseinc/linters-config/blob/master/react-redux/.github/workflows/linters.yml) instead.

**Comment by chiztechnology** [comment]
Thank you ! Let me try with this one

File: src/components/Books.js [hunk 1]
```diff
@@ -2,32 +2,26 @@ import React, { useEffect, useState } from 'react';
...
 
     const handleSubmit = (e) => {
         e.preventDefault();
-        dispatch(addBook({ id: Date.now(), ...book }));
+        console.log(book);
+        dispatch(saveBook({ id: Date.now(), ...book }));
 
         // clear input fields after dispatch
         setBook({ title: '', author: '' });
     };
```
Comment (levy002) [inline src/components/Books.js:19]: - [OPTIONAL] In order to make sure that the form inputs fields are cleared after form submission, please make sure to add the `value` attribute to each input and pass the state as a value
  **_Suggestion:_**
   `<input
              type='text'
              placeholder='Book title'
              name='title'
              onChange={(e) => setBook({ ...book, title: e.target.value })}
              value={book.title} 
            />
            <input
              type='text'
              placeholder='Author'
              name='author'
              onChange={(e) => setBook({ ...book, author: e.target.value })}
              value={book.author}
            />`

 - [OPTIONAL] Kindly make sure to add the validation so the user can not be able to add the empty book (without title and author).

**Review by levy002 — APPROVED** [review]
## Hello @chiztechnology :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Status: APPROVED :fire:  :fire: 

## Highlights
- Professional PR description :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books.js (hunk 1 above)
Comment (chiztechnology) [reply src/components/Books.js:19]: thanks you for your comment.
I'll add those changes to improve my project.

Best regards

**Pull request merged**